index.d.ts

```diff
@@ -0,0 +1,80 @@
+import { EventEmitter } from "events";
+import { User, Guild, Channel, Client, ClientOptions } from "eris";
+
+declare namespace ErisSharder {
+  interface Sharder extends EventEmitter {
+    shardCount: number;
+    firstShardID: number;
+    lastShardID: number;
+    clusterCount: number;
+    clusterTimeout: number;
+    token: string | false;
+    clusters: Map<>;
+    workers: Map<>;
+    queue: ErisSharderQueue;
+    eris: Client;
+    options: {
+      stats: ErisSharderStats;
+    };
+    statsInterval: number;
+    mainFile: unknown;
+    name: string;
+    guildsPerShard: number;
+    webhooks: {
+      cluster?: unknown;
+      shard?: unknown;
+    };
+    debug: boolean;
+    clientOptions: ClientOptions;
+		callbacks: Map<>;
+		isMaster(): boolean;
+		startStats(): void;
+		executeStats(clusters: unknown[], start: number): void;
+		start(clusterID: number): void;
+		launch(): void;
+		chunk(shards: unknown, clusterCount: number): unknown[]
+		connectShards(): void;
+		sendWebhook(type: string, embed): void;
+		printLogo(): void;
+		restartCluster(worker: unknown, code: unknown): void;
+		calculateShards(): Promise<unknown>;
+		fetchInfo(start: number, type: unknown, value: unknown): void;
+		broadcast(start: number, message: unknown): void;
+		sendTo(cluster: unknown, message: unknown): void;
+  }
+
+  interface Base {
+    bot: boolean;
+    clusterID: number;
+    ipc: ErisSharderIPC;
+    restartCluster(clusterID: number): void;
+  }
+
+  interface ErisSharderIPC extends EventEmitter {
+    events: Map<string, ErisSharderEvent>;
+    register(event: string, callback: Function): void;
+    unregister(name: string): void;
+    broadcast(name: string, message?: unknown): void;
+    sendTo(cluster: number, name: string, message?: unknown): void;
+    fetchUser(id: string): Promise<User>;
+    fetchGuild(id: string): Promise<Guild>;
+    fetchChannel(id: string): Promise<Channel>;
+  }
+
+  interface ErisSharderEvent {
+    fn: (msg: unknown) => void;
```

Comment:
Object

Reply:
Using `Object` type in TS is basically useless and will cause more harm than good. If you mean that it has to be some object like `{}` then the way to type it is to type the allowed values in that object in detail **OR** leave it as `unknown`.

Reply:
This is the Object type:
```ts
interface Object {
  // ...

  /** Returns a string representation of an object. */
  toString(): string;

  /** Returns a date converted to a string using the current locale. */
  toLocaleString(): string;

  /** Returns the primitive value of the specified object. */
  valueOf(): Object;

  /**
   * Determines whether an object has a property with the specified name.
   * @param v A property name.
   */
  hasOwnProperty(v: string): boolean;

  /**
   * Determines whether an object exists in another object's prototype chain.
   * @param v Another object whose prototype chain is to be checked.
   */
  isPrototypeOf(v: Object): boolean;

  /**
   * Determines whether a specified property is enumerable.
   * @param v A property name.
   */
  propertyIsEnumerable(v: string): boolean;
}```